grails-app/services/au/org/ala/spatial/portal/HubWebService.groovy

```diff
@@ -78,17 +78,27 @@ class HubWebService {
 
         try {
             if (type == HttpPost.METHOD_NAME) {
-                call = new PostMethod()
+                call = new PostMethod(url)
                 if (inputStream) {
                     call.setRequestEntity(new InputStreamRequestEntity(inputStream))
                 }
             } else {
                 //GET
-                call = new GetMethod()
+                call = new GetMethod(url)
             }
             if (contentType) {
                 call.setRequestHeader(HttpHeaders.CONTENT_TYPE, contentType)
             }
+            //Todo Test of supporting UserPrincipal
+            def user = authService.userId
+            if (user) {
```

Comment:
- how if grailsApplication.config.app.http.header.userId not exists?
- should we also save 'apiKey' into a similar config?

Reply:
1, header.userId is stored in application.grooy, so won't be missing. 
2, In Todo, this block is for UserPrincipal,  it aims  not to use apiKey